submissions/kotlyar-andrey/oop_exercise/index.js

```diff
@@ -0,0 +1,104 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/kotlyar-andrey/a-tiny-JS-world
+   Web app: https://kotlyar-andrey.github.io/a-tiny-JS-world/
+   */
+
+class Creature {
+  constructor(species, name, gender, phrase) {
+    this._species = species;
+    this._name = name;
+    this._gender = gender;
+    this._phrase = phrase;
+    this._friends = [];
+    this._properties = ["_species", "_name", "_gender", "saying"];
+  }
+
+  getFriends() {
+    return this._friends && this._friends.length > 0
+      ? this._friends.map((friend) => friend._name).join(", ")
+      : "no friends";
+  }
+
+  setFriends(newFriends) {
+    this._friends = [...newFriends];
+  }
+
+  getSaying() {
+    return self._phrase;
+  }
+
+  setSaying(newPhrase) {
+    this._phrase = newPhrase;
+  }
+
+  addProperties(...newProperties) {
+    this._properties = [...this._properties, ...newProperties];
+  }
+
+  display() {
+    const friends = this.getFriends();
+    print(
+      this._properties
+        .map((prop) => this[prop])
+        .join("; ")
+        .concat(`; ${friends}`)
+    );
+  }
+}
+
+class Animal extends Creature {
+  constructor(species, name, gender, phrase, legs, hands) {
+    super(species, name, gender, phrase);
+    this._legs = legs;
+    this._hands = hands;
```

Comment:
Do all animals have hands?
Imagine we popultae the world with birds, fish, rhinos and deer.
OOP helps to model the real world quite accurately avoiding defining properties and behavior that the counterparts in the real world do not have.